fili-core/src/main/java/com/yahoo/bard/webservice/web/util/PaginationParameters.java

```diff
@@ -90,6 +93,11 @@ private int parseParameter (String parameter, String parameterName) throws BadPa
             throw new BadPaginationException(errorMessage.format(parameterName));
         }
         try {
+            if (parameter.equals(FIRST)) {
+                return 1;
+            } else if (parameter.equals(LAST)) {
+                return LAST_PAGE;
```

Comment:
I think it might be better to make pagination parameters use an `Either<Integer, PageResultsFlag>`. I know this is only really needed for the `LAST` page case, but using `-1` as a special flag could lead to some confusion later.
```java
enum PageResultsFlag {
    FIRST("first"),
    LAST("last");
    // You could also have a function to handle #getPage(resultSize) in here for both cases
}
```